tests/ios/pages/ios_course_home_tab.py

```diff
@@ -0,0 +1,69 @@
+"""
+    Course Home Page Module
+"""
+
+from tests.ios.pages.ios_base_page import IosBasePage
+from tests.common import values
+from selenium.common.exceptions import NoSuchElementException
+
+
+class IosCourseHomeTab(IosBasePage):
+    """
+    Course Home Tab screen
+    """
+
+    def component_navigation(self, retries=5):
+        """
+        Navigate between components
+
+        Args:
+            retries (int): Maximum number of retries to find the finish button.
+
+        Returns:
+            webdriver element: finish Element
+        """
+        if retries == 0:
+            raise Exception("Finish button not found after maximum retries")
+
+        self.get_next_btn().click()
+
+        if self.get_finish_button():
+            return self.get_finish_button()
+        else:
+            return self.component_navigation(retries=retries - 1)
```

Comment:
Can we do wait for element or any other thing rather than this recursion?

Reply:
@jawad-khan 
I used this to check the Finish element appearance dynamically, it will click the Next button until the finish element appears, 
and due to this recursion, it will not stuck into any infinite loop in case of any internet issue or other exception.